CSharp/Library/Microsoft.Bot.Builder.Azure/DocumentDbBotDataStore.cs

```diff
@@ -253,16 +254,20 @@ internal DocDbBotDataEntity(IAddress key, BotStoreType botStoreType, BotData bot
 
         public static string GetEntityKey(IAddress key, BotStoreType botStoreType)
         {
+            string result;
             switch (botStoreType)
             {
                 case BotStoreType.BotConversationData:
-                    return $"{key.ChannelId}:conversation{key.ConversationId.SanitizeForAzureKeys()}";
+                    result = $"{key.ChannelId}:conversation{key.ConversationId.SanitizeForAzureKeys()}";
+                    return result.Substring(0, Math.Min(result.Length, MAX_KEY_LENGTH));
```

Comment:
>.Substring(0, Math.Min(result.Length, MAX_KEY_LENGTH)) [](start = 33, length = 54)

instead of just truncating, we should generate hashcode off of original string and make that end of the 254 char key.  That way we reduce chance of 2 keys colliding